Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -226,7 +226,7 @@ void GridMasonryLayout::updateRunningPositions(const RenderBox& child, const Gri
 
 void GridMasonryLayout::updateItemOffset(const RenderBox& child, LayoutUnit offset)
 {
-    m_itemOffsets.add(&child, offset);
+    m_itemOffsets.set(&child, offset);
```

Comment:
We could get rid of this function and move the single line to line 224.

It is only called in one place.

Reply:
The fact that this behavior is expected is shocking to me! Nice find.

I think you should keep this helper function -- it's a clear statement of what you are doing